assembl/static2/js/app/components/form/LoadSaveReinitializeForm.jsx

```diff
@@ -56,17 +59,29 @@ export default class LoadSaveReinitializeForm extends React.Component<Props, Sta
     });
   };
 
-  save = async (values: TInitialValues) => {
+  runMutations = async (values: TInitialValues) => {
     const { createMutationsPromises, save, afterSave } = this.props;
-    if (this.state.initialValues) {
-      const mutationPromises = createMutationsPromises(values, this.state.initialValues);
-      const status = await save(mutationPromises);
-      if (status === 'OK') {
-        // we really need to do a refetch to have the correct new ids in values
-        await this.load('network-only');
-        if (afterSave) afterSave(values);
+    // $FlowFixMe flow doesn't see that initialValues has already been checked
+    const mutationPromises = createMutationsPromises(values, this.state.initialValues);
+    const status = await save(mutationPromises);
+    if (status === 'OK') {
+      // we really need to do a refetch to have the correct new ids in values
+      await this.load('network-only');
+      if (afterSave) afterSave(values);
+    }
+  };
+
+  save = (values: TInitialValues) => {
+    const { withWarningModal, warningValues, warningMessageKey } = this.props;
+    const { initialValues } = this.state;
+    if (initialValues) {
+      const warningValuesHaveChanged = warningValues && warningValues.some(value => values[value] !== initialValues[value]);
+      if (withWarningModal && warningValuesHaveChanged && warningMessageKey) {
+        return displayConfirmationModal(values, this.runMutations, warningMessageKey);
       }
+      return this.runMutations(values);
     }
+    return undefined; // TODO: return an Error ?
```

Comment:
@paolina92 @pascalhuynh What would you do here ? The function needs to return something even when there are no initialValues. I could turn the eslint-warning off but it would be buggy

Reply:
`Undefined` fait généralement référence à quelque chose qui n'a pas encore reçu de valeur. Sinon, c'est `null`.

Sinon ?
```
try {
const result = withWarningModal && warningValuesHaveChanged && warningMessageKey ? displayConfirmationModal() : this.runMutations()
return result;
}
catch(error) {
  affiche une modale avec error
}
```

Reply:
Oui j'ai juste mis undefined pour me rappeler que je dois le changer. Là le problème c'est simplement que la fonction save doit retourner quelque chose si initialValues est undefined, il s'agit pas de la logique de l'affichage de la modale.

Reply:
return false ?